external/src/main/java/com/redhat/cloud/policies/engine/process/Receiver.java

```diff
@@ -72,105 +89,104 @@
-                    }
-                    input.ack();
-                    return null;
-                });
+        incomingMessagesCount.inc();
+        if (log.isTraceEnabled()) {
+            log.tracef("Received message, input payload: %s", input.getPayload());
+        }
+        JsonObject json;
+        try {
+            json = new JsonObject(input.getPayload());
+        } catch(Exception e) {
+            processingErrors.inc();
+            return input.ack();
+        }
+        if (json.containsKey(TYPE_FIELD)) {
+            String eventType = json.getString(TYPE_FIELD);
+            if(!ACCEPTED_TYPES.contains(eventType)) {
+                if (log.isDebugEnabled()) {
+                    log.debugf("Got a request with type='%s', ignoring ", eventType);
+                }
+                rejectedCount.inc();
+                return input.ack();
+            }
+        }
+
+        if (json.containsKey(HOST_FIELD)) {
+            json = json.getJsonObject(HOST_FIELD);
+        } else {
+            return input.ack();
+        }
+
+        // Verify host.reporter (not platform_metadata.metadata.reporter!) is one of the accepted values
+        String reporter = json.getString(REPORTER_FIELD);
+        if(!ACCEPTED_REPORTERS.contains(reporter)) {
+            rejectedCount.inc();
+            return input.ack();
+        }
+
+        String insightsId = json.getString(INSIGHT_ID_FIELD);
+
+        if (isEmpty(insightsId)) {
+            rejectedCount.inc();
+            return input.ack();
+        }
+
+        String tenantId = json.getString(TENANT_ID_FIELD);
+        String displayName = json.getString(DISPLAY_NAME_FIELD);
+        String text = String.format("host-egress report %s for %s", insightsId, displayName);
+
+        Event event = new Event(tenantId, UUID.randomUUID().toString(), INSIGHTS_REPORT_DATA_ID, CATEGORY_NAME, text);
+        // Indexed searchable events
+        Multimap<String, String> tagsMap = parseTags(json.getJsonArray(TAGS_FIELD));
+        tagsMap.put(DISPLAY_NAME_FIELD, displayName);
+        tagsMap.put(INVENTORY_ID_FIELD, json.getString(HOST_ID));
+        event.setTags(tagsMap);
+
+        // Additional context for processing
+        Map<String, String> contextMap = new HashMap<>();
+        contextMap.put(INSIGHT_ID_FIELD, insightsId);
+        event.setContext(contextMap);
+
+        JsonObject sp = json.getJsonObject(SYSTEM_PROFILE_FIELD);
+        Map<String, Object> systemProfile = parseSystemProfile(sp);
+
+        systemProfile.put(FQDN_NAME_FIELD, json.getString(FQDN_NAME_FIELD));
+
+        event.setFacts(systemProfile);
+
+        try {
+            List<Event> eventList = new ArrayList<>(1);
+            eventList.add(event);
+            if (storeEvents) {
+                alertsService.addEvents(eventList);
+            } else {
+                alertsService.sendEvents(eventList);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Need to `input.ack()` before?

Reply:
No, this is internal error - we want the message to be reprocessed from Kafka.

Reply:
Ok. Could you please add a comment about this then?